libraries/Microsoft.Bot.Builder.Dialogs.Rules/Rules/UtteranceRecognizeRule.cs

```diff
@@ -66,7 +66,7 @@ protected override PlanChangeList OnCreateChangeList(PlanningContext planning, o
                         this.Intent,
                     },
                     EntitiesMatched = this.Entities,
-                    EntitiesRecognized = entitiesRecognized,
+                    TurnState = recognizedEntities,
```

Comment:
When I suggested moving to the turnState, I meant moving to using the turn state to save the entities, not to replace the turn state with the entities. So, my recommendation is to use the TurnContext Get and Set methods to save and get the recognized entities. Does that make sense?

Reply:
After debuging, I changed the name back to EntitiesRecognized. Because there are differences on memories between C# and Node. In node, the PlanningChangeList has a TurnState property, which is got from memory.
![image](https://user-images.githubusercontent.com/32191031/55067911-89a66280-50bb-11e9-9603-69ef71d86cb6.png)
![image](https://user-images.githubusercontent.com/32191031/55067987-ba869780-50bb-11e9-85dc-094d367dd45f.png)

But in C# we don't have this step yet, and also the TurnState usage in C# and Node is not same.

It will break if I use TurnState here, so I changed back to EntitiesRecognized now to replace the Entities with Turn. 
We should have other changes for the memory stuff. 
Please let me know your comments.

Thanks,
Lu